internal/command/init.go

```diff
@@ -1031,23 +1031,87 @@ func (c *InitCommand) backendConfigOverrideBody(flags arguments.FlagNameValueSli
 			flushVals() // deal with any accumulated individual values first
 			mergeBody(newBody)
 		} else {
+			// The flag value is setting a single attribute's value
 			name := item.Value[:eq]
 			rawValue := item.Value[eq+1:]
-			attrS := schema.Attributes[name]
-			if attrS == nil {
+
+			splitName := strings.Split(name, ".")
+			isNested := len(splitName) > 1
+
+			var value cty.Value
+			var valueDiags tfdiags.Diagnostics
+			switch {
+			case !isNested:
+				// The flag item is overriding a top-level attribute
+				attrS := schema.Attributes[name]
+				if attrS == nil {
+					diags = diags.Append(tfdiags.Sourceless(
+						tfdiags.Error,
+						"Invalid backend configuration argument",
+						fmt.Sprintf("The backend configuration argument %q given on the command line is not expected for the selected backend type.", name),
+					))
+					continue
+				}
+
+				value, valueDiags = configValueFromCLI(item.String(), rawValue, attrS.Type)
+				diags = diags.Append(valueDiags)
+				if valueDiags.HasErrors() {
+					continue
+				}
+
+				// Synthetic values are collected as we parse each flag item
+				synthVals[name] = value
+			case isNested:
+				// The flag item is overriding a nested attribute
+				// e.g. assume_role.role_arn in the s3 backend
+				// We assume a max nesting-depth of 1 as s3 is the only
```

Comment:
Can we somehow bake this assumption in the code validation?

An input like `"test_nesting_single.child.grand"` was allowed to pass with no error, and the child value ended up being set.

Reply:
I've addressed this by swapping to a solution that allows any level of nesting, and that change [also addresses issues related to missing parent fields](https://github.com/hashicorp/terraform/pull/36919#discussion_r2057616146). I figured this was a better approach, as the alternative was to add an error asking users to report issues with overriding fields at a greater level of nesting, and why kick that issue down the road?